segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -142,6 +146,30 @@ public void setupAppend(SetupAppend setupAppend) {
                 tokenVerifier.verifyToken(newSegment,
                         setupAppend.getDelegationToken(),
                         AuthHandler.Permissions.READ_UPDATE);
+
+                final Duration durationToExpiry = JwtUtils.durationToExpiry(setupAppend.getDelegationToken());
+
+                if (durationToExpiry != null) { // Can be null if token had no expiry set (for internal communications)
+                    if (durationToExpiry.isNegative()) {
+                        log.debug("Token has already expired");
+                        throw new TokenExpiredException("Token already expired");
+                    } else {
+                        Futures.delayedTask(() -> {
+                            if (writerStates.containsKey(Pair.of(newSegment, writer))) {
+                                // Closing the connection will result in client authenticating with Controller again
+                                // and retrying.
+                                log.debug("Closing client connection due to token expiry");
+                                try {
+                                    this.connection.close();
```

Comment:
if connection pooling is enabled then all the sessions will be terminated. We should send a `WireCommands.AuthTokenCheckFailed` .

Reply:
Sure, I've made the change in the latest commits that I just pushed.